src/style/types/attribute.rs

```diff
@@ -163,7 +174,10 @@ impl Attribute {
     /// Returns the SGR attribute value.
     ///
     /// See <https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters>
-    pub fn sgr(self) -> i16 {
-        SGR[self as usize]
+    pub fn sgr(self) -> String {
+        if (self as usize) > 4 && (self as usize) < 9 {
+            return "4:".to_string() + SGR[self as usize].to_string().as_str()
```

Comment:
This function needs to return a string, because the code to set attributes isn't the same for the other types of underline. In the function, I'm just checking if the attribute is one that should be printed differently.

Reply:
It basically returns `4:{sgr}` for the other underlines and `{sgr}` for every other attribute.